### PR TITLE
Proxy retry all the things redux

### PR DESCRIPTION
**Pull request by Fizzadar**
Rework of https://github.com/mautrix/instagram/pull/103 that moves the retry logic into the base `AndroidAPI` class meaning everything will be wrapped.

### CODE REVIEW

File: mauigpapi/http/base.py [hunk 1]
```diff
@@ -199,7 +208,15 @@ async def std_http_get(
     ) -> T:
         headers = {**self._headers, **headers} if headers else self._headers
         query = {k: v for k, v in (query or {}).items() if v is not None}
-        resp = await self.http.get(url=self.url.with_path(path).with_query(query), headers=headers)
+        url = self.url.with_path(path).with_query(query)
+        resp = await proxy_with_retry(
+            f"AndroidAPI.std_htt_get: {url}",
```
Comment (tulir) [inline mauigpapi/http/base.py:213]:
```suggestion
            f"AndroidAPI.std_http_get: {url}",
```

**Review by tulir — APPROVED** [review]

**Comment by Fizzadar** [comment]
Sorry @tulir couple of missed calls to the raw API I just added.

File: mautrix_instagram/portal.py [hunk 2]
```diff
@@ -870,26 +870,29 @@ async def convert_to_ogg(data, mimetype):
...
-                    f"{url} was too large ({length} > {self.matrix.media_config.upload_size})"
-                )
-                raise ValueError("Attachment not available: too large")
-            data = await resp.read()
-            if not data:
-                return None, ""
-            mimetype = resp.headers["Content-Type"] or magic.from_buffer(data, mime=True)
-            return data, mimetype
+        async def download():
+            async with source.client.raw_http_get(url) as resp:
```
Comment (tulir) [inline mautrix_instagram/portal.py:874]: We should figure out lower timeouts for `raw_http_get` at some point so media wouldn't get stuck for too long (doesn't necessarily have to be in this PR)

**Review by tulir — APPROVED** [review]

File: mautrix_instagram/portal.py (hunk 2 above)
Comment (Fizzadar) [reply mautrix_instagram/portal.py:874]: Yep, was thinking another PR for that, let's get this out :)

**Pull request merged**